src/controller/AutoCommissioner.cpp

```diff
@@ -927,77 +967,19 @@ CHIP_ERROR AutoCommissioner::PerformStep(CommissioningStage nextStage)
     return CHIP_NO_ERROR;
 }
 
-void AutoCommissioner::ReleaseDAC()
-{
-    if (mDAC != nullptr)
-    {
-        Platform::MemoryFree(mDAC);
-    }
-    mDACLen = 0;
-    mDAC    = nullptr;
-}
-
-CHIP_ERROR AutoCommissioner::SetDAC(const ByteSpan & dac)
-{
-    if (dac.size() == 0)
-    {
-        ReleaseDAC();
-        return CHIP_NO_ERROR;
-    }
-
-    VerifyOrReturnError(dac.size() <= Credentials::kMaxDERCertLength, CHIP_ERROR_INVALID_ARGUMENT);
-    if (mDACLen != 0)
-    {
-        ReleaseDAC();
-    }
-
-    VerifyOrReturnError(CanCastTo<uint16_t>(dac.size()), CHIP_ERROR_INVALID_ARGUMENT);
-    if (mDAC == nullptr)
-    {
-        mDAC = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(dac.size()));
-    }
-    VerifyOrReturnError(mDAC != nullptr, CHIP_ERROR_NO_MEMORY);
-    mDACLen = static_cast<uint16_t>(dac.size());
-    memcpy(mDAC, dac.data(), mDACLen);
-    mParams.SetDAC(ByteSpan(mDAC, mDACLen));
-
-    return CHIP_NO_ERROR;
-}
-
-void AutoCommissioner::ReleasePAI()
+CHIP_ERROR AutoCommissioner::AllocateMemoryAndCopySpan(Platform::ScopedMemoryBufferWithSize<uint8_t> & scopedBuffer, ByteSpan span)
 {
-    if (mPAI != nullptr)
+    if (!span.size())
     {
-        chip::Platform::MemoryFree(mPAI);
+        return CHIP_ERROR_INVALID_MESSAGE_LENGTH;
```

Comment:
So for JF there is always required to be an ICAC, I guess?

Reply:
That's right, an ICAC is mandatory for JF.